examples/remote/binary_dependencies/BUILD.bazel

```diff
@@ -8,7 +9,5 @@ rust_binary(
     data = [
         "//remote/binary_dependencies/cargo:cargo_bin_texture_synthesis",
     ],
-    deps = [
-        "//remote/binary_dependencies/cargo:ferris_says",
-    ],
+    deps = all_crates(),
```

Comment:
Why is this change being made? Do you think end users should just include "all crates"?

I'm against this because I'm generally against making dependencies opaque anywhere. libraries and binaries should include what they need and only what they need.

Reply:
> Why is this change being made? Do you think end users should just include "all crates"?

Here I'm demonstrating what is possible with the new macros, but yes, I think it's nicer to be able to simply refer to your dependencies via `all_crates` rather than having to track your dependencies twice, once in the `Cargo.toml` file and again in your BUILD files. I also don't think it's unclear what dependencies are being used at build time since the way `cargo-raze` functions is everything is written out before Bazel interracts with it. Any confusion can be resolved by reading the `crates.bzl` file and looking for your package name in the maps. I generally agree with the value of being explicit about dependencies, but I don't think that philosophy plays nicely with a tool that generates dependencies.